lib/ruby_lsp/listeners/spec_style.rb

```diff
@@ -0,0 +1,200 @@
+          dispatcher: Prism::Dispatcher,
+          uri: URI::Generic,
+        ).void
+      end
+      def initialize(response_builder, global_state, dispatcher, uri)
+        @response_builder = response_builder
+        @current_group = T.let(nil, T.nilable(Requests::Support::TestItem))
+        @uri = uri
+        @index = T.let(global_state.index, RubyIndexer::Index)
+
+        @visibility_stack = T.let([:public], T::Array[Symbol])
+        @nesting = T.let([], T::Array[String])
+        @test_group_nesting = T.let([], T::Array[String])
+        @spec_stack = T.let([], T::Array[T::Boolean])
+
+        dispatcher.register(
+          self,
+          :on_class_node_enter,
+          :on_class_node_leave,
+          :on_module_node_enter,
+          :on_module_node_leave,
+          :on_call_node_enter,
+          :on_call_node_leave,
+          :on_def_node_enter,
+        )
+      end
+
+      sig { params(node: Prism::ClassNode).void }
+      def on_class_node_enter(node)
+        @visibility_stack << :public
+        name = constant_name(node.constant_path)
+        name ||= name_with_dynamic_reference(node.constant_path)
+
+        fully_qualified_name = RubyIndexer::Index.actual_nesting(@nesting, name).join("::")
+
+        attached_ancestors = begin
+          @index.linearized_ancestors_of(fully_qualified_name)
+        rescue RubyIndexer::Index::NonExistingNamespaceError
+          # When there are dynamic parts in the constant path, we will not have indexed the namespace. We can still
+          # provide test functionality if the class inherits directly from Test::Unit::TestCase or Minitest::Test
+          [node.superclass&.slice].compact
+        end
+
+        @spec_stack.push(attached_ancestors.include?("Minitest::Spec"))
+
+        if attached_ancestors.include?("Minitest::Spec")
+          @response_builder.add(Requests::Support::TestItem.new(
+            fully_qualified_name,
+            fully_qualified_name,
+            @uri,
+            range_from_node(node),
+          ))
+        end
+
+        @nesting << name
+      end
+
+      sig { params(node: Prism::ModuleNode).void }
+      def on_module_node_enter(node)
+        @visibility_stack << :public
+
+        name = constant_name(node.constant_path)
+        name ||= name_with_dynamic_reference(node.constant_path)
+
+        @nesting << name
+      end
+
+      sig { params(node: Prism::ModuleNode).void }
+      def on_module_node_leave(node)
+        @visibility_stack.pop
+        @nesting.pop
+      end
+
+      sig { params(node: Prism::ClassNode).void }
+      def on_class_node_leave(node)
+        @visibility_stack.pop
+        @nesting.pop
+        @spec_stack.pop
+      end
+
+      sig { params(node: Prism::CallNode).void }
+      def on_call_node_enter(node)
+        case node.name
+        when :describe
+          description = handle_example_or_group(node)
+
+          @test_group_nesting << description if description
+        when :it
```

Comment:
Does minitest-spec have any other kind of syntax we need to consider? (I know that RSpec has `specify` as an alias for `it`).